app/models/episode.rb

```diff
@@ -263,10 +263,11 @@ def update_contents(files)
 
   def find_existing_content(pos, url)
     return nil if url.blank?
-    content_file = URI.parse(url || '').path.split('/').last
+    content_file = URI.parse(url || '').path.split('/')[-2, 2].join('/')
```

Comment:
Also noting that the `url || ''` isn't actually necessary.  And if url ever was blank, that would throw an error.